canonical-websites/snapcraft.io
# Update API module with caching on public pages

Update the cache module to a more generic requests module (open to other naming ideas).

I modified this code to extend the requests session object with some custom code to manage our custom API exceptions. It includes a factory to set the session up and I left the `get` function available, which uses the new session in the background.

I updated the Public API modules to use the cached session. In turn I simplified them taking advantage of the Requests session API. Setting default headers and removing the duplicate code.

## QA

To disable internet to test, you can turn your internet off or you can modify the `webapp/api/public.py` file. If you change the URL on line 11 to `'https://google.com:81',` all the API traffic will time out.

 - `./run test` - All should be OK (As it is in the CI test)
 - `./run` and try the site in general, it should all work well.
 - Go to http://localhost:8004/search?q=toto and see some results.
 - Disable the internet and still see the results
 - Restart with `./run --env WEBAPP=lime`
 - Go to the same URL and there should be no results.
 - Turn the internet back on and still no results but searching for "lime" returns results.
